api/include/opentelemetry/trace/span.h

```diff
@@ -158,12 +159,13 @@ class Span
 
   // TODO
   // SpanContext context() const noexcept = 0;
-
+  virtual trace::SpanContext GetContext() const noexcept = 0;
   // Returns true if this Span is recording tracing events (e.g. SetAttribute,
   // AddEvent).
   virtual bool IsRecording() const noexcept = 0;
 
-  virtual Tracer &tracer() const noexcept = 0;
+  // Cannot have Tracer in Span right now because it will cause cyclical dependency issue
```

Comment:
This comment belongs instead in the PR description to address why Span and it's descendants have tracer() removed.

Reply:
I have remove these lines and moved the reasoning to the PR description